src/log/message.ts

```diff
@@ -18,6 +18,13 @@ import {Type} from '../type';
 import {stringify} from '../util';
 import {VgSortField} from '../vega.schema';
 
+export function dropChannelOnMark(mark: Mark, channel: Channel) {
+  return `${mark} does not support ${channel}: dropping ${channel}`;
+}
+
+export const ROUNDED_CORNER_STACKED_BAR_WITH_AVOID =
+  'Labels on stacked bar chart can only avoid the base mark (stacked bars)';
```

Comment:
Can you explain why we have this limitation? 

Does this mean that labels stacked bar charts also can't avoid "all marks"?

Reply:
There is a typo here. This limitation only applies to stack bar charts with rounded corners.
In Vega, a label mark can only avoid other marks that comes before itself.
We have this limitation because Vega-Lite compiles a stack bar chart with rounded corners into a group mark. So, we have to put the label mark inside the group mark. For this reason, we cannot pull the label mark out to avoid other marks that comes after the stacked bar chart.
On the other hand, a regular stacked bar chart is just a normal bar chart without grouping. So, we can push the label mark to be behind all the other marks to avoid them.